Sources/ComposableArchitecture/Store.swift

```diff
@@ -362,15 +362,23 @@ public final class Store<State, Action> {
     self.scope(state: toLocalState, action: { $0 })
   }
 
-  func send(_ action: Action, originatingFrom originatingAction: Action? = nil) {
+  func send(_ action: Action, originatingFrom originatingAction: Action? = nil, instrumentation: Instrumentation = .shared) {
     self.threadCheck(status: .send(action, originatingAction: originatingAction))
 
     self.bufferedActions.append(action)
     guard !self.isSending else { return }
 
     self.isSending = true
     var currentState = self.state.value
+
+    let callbackInfo = Instrumentation.CallbackInfo(storeKind: self.self, action: action, originatingAction: originatingAction).eraseToAny()
```

Comment:
I think it's important to suppress CallbackInfo construction if instrumentation.callback == nil.
Could do that with a method Instrumentation.callbackInfo() that returned nil if callback is nil, and mark the CallbackInfo ctor itself file-private to ensure callsites like this one here are converted to using the new helper.

Reply:
That would ripple out to the callback then needing to take a `CallbackInfo?` type, introducing a wrapper function around the callback closure that also checks to null-ness of the info and callbacks themselves:

```swift
func callbackInfo(storeKind: ..., action: ..., originatingAction: ...) -> CallbackInfo? {
  guard callback != nil else { return nil }
  ...
}

func callback(info: CallbackInfo?, when: ..., what: ...) {
  guard let info = info, let callback = callback else { return }
  callback(info, when, what)
}
```

Reply:
Per tuple: for the time being we will land as is. Can investigate moving this into a helper method if we decided to attempt upstreaming.